### ISSUE TITLE
Setting reasonable default values

### ISSUE THREAD

**Issue by huang195** [iter8-tools/iter8#911]
**Is your feature request related to a **problem**? Please describe the **problem**.**
A difficulty user might face when using Iter8 is when creating an Experiment CR as the corresponding CRD is very complicated. We should make a minimal viable working Experiment CR as simple as possible, as few lines as possible, and internally fill in default values for users.

**Describe the feature/solution you'd like**
Couple of implementation options:
1. As part of existing etc3 functionality to fill in the defaults
2. Add mutating webhook function to etc3 to fill in the defaults

**Will this feature/solution bring new benefits for Iter8 users? What are they?**
Yes, makes using Iter8 easier for new users.

cc: @sriumcp @kalantar @fabolive

**Comment by sriumcp** [comment]
In addition to #862 (auto generate names for versions) and #863 (auto generate testing pattern) , what are some additional defaults we might want to set (either as part of the current etc3 defaulting, or in a webhook)?

**Comment by huang195** [comment]
I'm thinking the following might be useful:
- defaulting handler tasks: e.g., check for targets to be ready
- defaulting metrics
- defaulting timing parameters, e.g., duration, iteration, etc.
- defaulting weightObjRef: this might be harder as that would require runtime information, which we currently don't have the capabilities

Anything else that would make a working Experiment CR as close to a `one-liner` as possible.

**Comment by sriumcp** [comment]
1. Re: timing, we do have defaults currently already (not sure what the sanest defaults here could be).

Some examples for the other categories would be nice to see...

**Comment by sriumcp** [comment]
I think you brought up this or a variant of this idea earlier @huang195 ... anyway, here it goes ...

Make `spec.strategy` section optional. This is possible once we start auto generating testing pattern (#863).

At this point, everything in the spec section other than target is optional, which will enable the user to specify a valid experiment that can be run by etc3 as follows.

```yaml
kind: experiment
apiVersion: iter8.tools
metadata:
  name: my-exp
spec:
  target: my-app
  # this is a non-breaking CRDChange
  experimentTemplate: namespace/name # for example `default/dummy`; optional
  # dummy is another experiment, with `iter8.tools/experiment.template: "true"` annotation set.
  # etc3 will not attempt to "run" this experiment, since it is a template
  # The spec section of `dummy` will be used as a default for the spec section of this experiment. 
  # Any information provided by the user in the current experiment will override the defaults 
  # defined in `dummy`.
  # In other words, the user only needs to specify the diff between this experiment and dummy.
  # It is always a good idea to copy over the defaults explicitly in the `status` section.
  # This makes it explicit what the values used in the current experiment are.
```

We do not want to make the target optional -- ensuring that it is required provides a critical advantage -- of sequencing experiments correctly for a given target. Besides it is a simple string and easy to set.

**Comment by sriumcp** [comment]
I feel setting sane defaults in the experiment across the board for every user/application, without help from the user, is not going to be possible (because no such sane default exists, given the breadth and scope of the Iter8 experiment CRD).

However, with the `dummy` template option described above, the experiment spec is substantially simplified for CI/CD/GitOps integration (you set the `dummy` template once at the beginning of time for an application, and re-use in each new experiment for the given target; of course, user will need to think through the sane defaults once per application).

**Comment by huang195** [comment]
Good point @sriumcp. Using only static information, one can only go so far filling in the default. Perhaps we can hold off on this issue until we have some runtime capabilities.

**Comment by kalantar** [comment]
In general, with defaulting at this scale, I agree that fixed defaults won't work. Rather, some form of template would be needed as suggested above. There is a certain benefit to defaulting per application. Use in CI/CD means that the experiment generated is easier to generate. I feel like templates combined with annotations (issue https://github.com/iter8-tools/iter8/issues/864) might make sense as a non-helm way to create experiments. It doesn't provide the means to create any application elements if needed though.

**Comment by sriumcp** [comment]
Subsumed by #948
